### ISSUE TITLE
inconsistent data member naming convention in pcl::PointCloud

### ISSUE THREAD

**Issue by csukuangfj** [PointCloudLibrary/pcl#2469]
## Your Environment

* PCL Version: the master branch

## Context

See the code
https://github.com/PointCloudLibrary/pcl/blob/9467b944ee3e30bb4cf8445d861c280f0fba1020/common/include/pcl/point_cloud.h#L413-L423

https://github.com/PointCloudLibrary/pcl/blob/9467b944ee3e30bb4cf8445d861c280f0fba1020/common/include/pcl/point_cloud.h#L592

Some data members are ended with `_` and some are not.
## Expected Behavior

Either all end with `_` or none.
## Current Behavior

The naming convention is not consistent.
## Code to Reproduce

See the code above.
## Possible Solution

Rename the data members consistently.

**Comment by taketwo** [comment]
You are right, there should be no `_` suffix. However, I don't see a backwards-compatible way to perform such a rename. Do you have any ideas?

**Comment by SergioRAgostinho** [comment]
Here's a suggestion. http://cpp.sh/6yamj

**Comment by csukuangfj** [comment]
@SergioRAgostinho 
Another solution might be
```.cpp
union
{
    int field;
    int field_;
};
```
which can handle the assignment operator and copy constructor automatically.

**Comment by SergioRAgostinho** [comment]
That's definitely a better strategy as performs similar.

@taketwo maybe we could even redo the same in #2102. Move that one to a simple deprecation.

**Comment by taketwo** [comment]
@csukuangfj great idea! And we can add a deprecation warning as well:

```cpp
union
{
    int field;
    int field_ __attribute__ ((deprecated));
};
```

@SergioRAgostinho I'm not sure... do you think it's worth the effort?

**Comment by csukuangfj** [comment]
https://github.com/PointCloudLibrary/pcl/blob/d1d59c63e572b88f8c48f006095732d66aae301e/common/include/pcl/pcl_macros.h#L324-L337

It's better use `PCL_DEPRECATED()` since it is compiler independent.

**Comment by kunaltyagi** [comment]
> do you think it's worth the effort

**Comment by csukuangfj** [comment]
> do you think it's worth the effort



Not sure. I haven't been using this project for quite a long time.
